### PR TITLE
Feature/90 endpoint verify etl success

### PR DESCRIPTION
**Pull request by cschwinderg**
## Related Issues:
* [EQ-90](https://jira.epa.gov/browse/EQ-90)
* [EQ-136](https://jira.epa.gov/browse/EQ-136)

## Main Changes:
* Added a new `etl_status` table for tracking the current status of etl (idle, running, failed, success). 
* Added code to the etl that updates the `etl_log`, `etl_schemas`, and `etl_status` tables if the etl failed because the server crashed. Sometimes the cloud.gov etl server crashes (ex. out of memory) the server then restarts, but the `etl_log` wouldn't have anything in the `load_error` column. This will work around this edge case.
* Added new endpoints: `health\etlDatabase`, `health\etlDomainValues`, and `health\etlGlossary`. These just verify the etl completed successfully within the desired time period. 
* Added a separate etl schedule for the glossary. Now the etl for the glossary runs every day, except Sunday, at 1 am. Sunday is excluded from the glossary etl, because we don't want to run the glossary etl twice on Sunday when the main process runs.

## Steps To Test:
1. Run `npm run start` from the etl folder to create the `etl_status` table
2. Verify the `etl_status` table was created and verify that all values are `idle`
3. Run `npm run etl_all` from the etl folder
4. Verify everything is loaded correctly and the values in `etl_status` are all `success`
5. Run `npm run start` from the app folder
6. Navigate to the following and verify the status returned is `UP`: 
    * http://localhost:9090/health/etlDatabase
    * http://localhost:9090/health/etlDomainValues
    * http://localhost:9090/health/etlGlossary
7. Update the statuses in the `etl_status` table to be `failed`
8. Navigate to the following and verify the status returned is `FAILED_DB`: 
    * http://localhost:9090/health/etlDatabase
    * http://localhost:9090/health/etlDomainValues
    * http://localhost:9090/health/etlGlossary
9. Simulate a server crash and restart by doing the following: 
    1. Run `npm run etl_all` from the etl folder
    2. Stop it part of the way through
    3. Run `npm run etl_all` from the etl folder again
10. Check the `etl_log` table and verify there is a record that says `Server crashed. Check cloud.gov logs for more info`.
11. Look at the etl schedules and verify the glossary cron job is set to run Mon - Sat at 1AM and the primary cron job is set to run Sunday at 1 AM.

### CODE REVIEW

File: etl/app/server/database.js [hunk 1]
```diff
@@ -421,7 +530,7 @@ export async function trimSchema(pool, s3Config) {
   }
 }
 
-// Drop old national-downloads folders
+// Get the ETL task for a particular profile
```
Comment (maxdiebold-erg) [inline etl/app/server/database.js:533]: Is this change intentional? It looks like the old comment is the correct description, and the new one is the same as the comment for `getProfileEtl`.

**Review by maxdiebold-erg — CHANGES_REQUESTED** [review]
This looks good! I made a couple suggestions, only one of which relates to a potential bug.

File: etl/app/server/database.js (hunk 1 above)
Comment (cschwinderg) [reply etl/app/server/database.js:533]: That was accidental. I'll switch back to the original comment.

File: etl/app/server/database.js [hunk 2]
```diff
@@ -103,6 +126,78 @@ export async function checkLogTables() {
...
+export async function checkForServerCrash() {
+  const client = await connectClient(eqConfig);
+
+  try {
+    // check the etl_status table
+    const result = await client
+      .query('SELECT * FROM logging.etl_status')
+      .catch((err) => log.warn(`Could not query databases: ${err}`));
+
+    if (!result.rowCount) {
```
[the remark this answers is not in the document]
Comment (cschwinderg) [reply etl/app/server/database.js:140]: Good catch!

**Review by maxdiebold-erg — APPROVED** [review]

**Comment by maxdiebold-erg** [comment]
Good to go!
